Jackhammers cannot tunnel through walls

RenechCDDA opened the pull request:
#### Summary
Balance "Jackhammers cannot tunnel through walls"

#### Purpose of change
I was reminded recently of how absurd this holdover is.

#### Describe the solution
Kill it. Jackhammers cannot tunnel through walls, at all. They can still be used against road surfaces(*why* would you do this? I don't know), boulders, and other MINEABLE terrain that isn't a wall.

~~The times were made less insane (going from a minimum of **20 minutes** to hand-tunnel through a m^3 of solid rock) to a minimum of 2 hours, for cases of mining 'downwards' with powered assistance. For hand picks the time went up to 8 hours, based largely on the AD&D1e numbers found in this stackexchange post: https://rpg.stackexchange.com/questions/141773/how-long-does-it-take-to-mine-rock~~

~~I have previously tried to find literature describing the tunnel at the siege of st andrews for a better real-world set of numbers, but I was unable. It's a relatively obscure siege from 500 years ago, and the tunneling was simply one part of it. So.... yeah. Would be happy to read for it if I can find it though.~~

Everything has been piped through dig_tool(), so we do not have two code paths.

While I was here I removed the strength scaling on pickaxes. There's no indication I can find that a miner's physical strength has any real effect on their ability to swing a pickaxe for hours on end. That's just endurance, which we already model through weariness.

#### Describe alternatives you've considered
This probably also wants a more comprehensive review of other problem areas where this cheese would be used (e.g. labs), but I don't want to do some huge audit right now.

#### Testing


#### Additional context

RenechCDDA committed:
Jackhammers cannot tunnel through walls

GuardianDll commented:
Do we want to mone down? That's like 4 meters down, i don't think you can do it in less than a full day, let alone week

GK-Edelweiss commented:
"Perhaps we shouldn't be so strict. I consulted an AI, and it suggested that a jackhammer can be a very effective tool for wall demolition."

John-Candlebury commented:
Yeah I dont think this is the required change, because jackhammers are frequently used in construction work to demolish concrete structural elements when doing renovations and the difference between that and mining through walls is just the game using words in a weird way. Using the jackhammer shouldnt be incredibly convenient to the point you'd choose it over other entry options if they are available, but jackhammer really are one of the most practical options if you need a small piece of concrete wall down.

It should take a time span of around 6-8 hours for wall being generous(meaning the player know how to do it) and be very strenous but removing it as an option isnt good.

RenechCDDA commented:
Andrei8L pointed out that this only affects the zone activity, and he is right. (Why do we have two separate code paths for this?)

So, gotta fix that before it's mergeable.

RenechCDDA committed:
Pipe everything through iuse / dig_tool()

Remove crafting helpers
Remove move adjustment based on strength
Remove bizarre misleading message based on bash resistance
Remove unreachable/impossible check for incorporeal effect

RenechCDDA commented:
Increasing the mining times has been pushed out to a later PR for separate review. Removal of strength scaling on pickaxing is still in this PR. I also removed an iuse-only check for crafting helpers and some other miscellanea, the commit message has more details.
